front/app/containers/Admin/index.tsx

```diff
@@ -106,7 +106,10 @@ const AdminPage = memo<Props & WithRouterProps>(
         clHistory.push('/');
       }
 
-      if (pathname.endsWith('/admin') || pathname.endsWith('/admin/')) {
+      if (
+        authUser &&
```

Comment:
Always need authUser in the admin, so moved the check to the front to reduce the need for further checks (since I made isAdmin stricter).